lite/memprovider.go

```diff
@@ -52,25 +54,64 @@ func (m *memStoreProvider) StoreCommit(fc FullCommit) error {
 	defer m.mtx.Unlock()
 	m.byHash[key] = fc
 	m.byHeight = append(m.byHeight, fc)
-	sort.Sort(m.byHeight)
+	m.sorted = false
```

Comment:
> Why not just sort it here?
The other performance boost comes from lazily sorting i.e when needed instead of everytime.

> In LatestCommit method we use byHeight but don't check if its sorted. Does it need to be ?
Wow, thanks for raising this. Shows that our tests ain't that comprehensive or that we don't have LatestCommit tests
```shell
$ grep -Rn 'LatestCommit(' *
client/provider.go:80:func (p *provider) LatestCommit() (fc lite.FullCommit, err error) {
client/provider.go:81:	commit, err := p.GetLatestCommit()
client/provider.go:91:func (p *provider) GetLatestCommit() (*ctypes.ResultCommit, error) {
client/provider_test.go:29:	seed, err := p.LatestCommit()
files/provider.go:105:func (p *provider) LatestCommit() (fc lite.FullCommit, err error) {
memprovider.go:131:func (m *memStoreProvider) LatestCommit() (FullCommit, error) {
provider.go:16:	LatestCommit() (FullCommit, error)
provider.go:90:func (c cacheProvider) LatestCommit() (fc FullCommit, err error) {
provider.go:93:		tfc, err = p.LatestCommit()
provider_test.go:30:func (missingProvider) LatestCommit() (lite.FullCommit, error) {
proxy/certifier.go:18:	fc, err := source.LatestCommit()
proxy/certifier.go:21:	fc, err := trust.LatestCommit()
proxy/query_test.go:64:	latest, err := source.LatestCommit()
```